astroid/interpreter/_import/spec.py

```diff
@@ -150,7 +175,16 @@ def find_module(
             if modname in sys.stdlib_module_names or (
                 processed and processed[0] in sys.stdlib_module_names
             ):
```

Comment:
I guess I'm still wondering if this can/should be:
```suggestion
            if processed and processed[0] in sys.stdlib_module_names:
```

Given that:
```sh
mkdir foo
pylint foo.io
```
Will unexpectedly evaluate True here.

Reply:
No, because then `collections` will never be considered frozen. We need both `collections` and `collections.abc` to pass this check.

Reply:
Got it -- then let's update the comment, since it sounds like we know whether something is a stdlib module if this test passes.

Reply:
We don't 😅 

The issue some reporter/primer encountered was that any module called `io` would also pass this check, which makes sense. It is just a `str` lookup. That's why I have changed the logic around now. If we can find the file we use that, if we can't (such as for frozen modules) we try that logic before giving up.

Reply:
Right, we don't -- but the comment makes it sound like we do. I added another comment with a suggestion.